tests/utils_tests/test_timezone.py

```diff
@@ -54,6 +57,24 @@ def test_now(self):
         with override_settings(USE_TZ=False):
             self.assertTrue(timezone.is_naive(timezone.now()))
 
+    def test_localdate(self):
+        naive = datetime.datetime(2015, 1, 1, 0, 0, 1)
+        with self.assertRaisesMessage(ValueError, 'astimezone() cannot be applied to a naive datetime'):
```

Comment:
Please check with Python 3.6, no exception is raised here (see a7a7ecd2b026c61a39a46d2d7eced0e06a92c970).